packages/react/src/types.ts

```diff
@@ -36,6 +36,8 @@ export type ComponentKeyboardEventHandler<TProps> = (
   data?: TProps,
 ) => void
 
+export type ComponentMouseEventHandler<TProps> = (event: MouseEvent, data?: TProps) => void
```

Comment:
Let's do not do this, it's to early for this change and it should be done separately for all components